pkg/networkservice/common/excludedprefixes/server.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package excludedprefixes provides a networkservice.NetworkServiceServer chain element that can read excluded prefixes
+// from config map and add them to request to avoid repeated usage.
+package excludedprefixes
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/trace"
+	"github.com/networkservicemesh/sdk/pkg/tools/prefixpool"
+)
+
+type excludedPrefixesService struct {
+	prefixes prefixpool.PrefixPool
+}
+
+// Note: request.Connection, Connection.Context and Context.IpContext should not be nil
+func (eps *excludedPrefixesService) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	logger := trace.Log(ctx)
+
+	conn := request.Connection
+	prefixes := eps.prefixes.GetPrefixes()
+	logger.Infof("ExcludedPrefixesService: adding excluded prefixes to connection: %v", prefixes)
+	ipCtx := conn.Context.IpContext
+	ipCtx.ExcludedPrefixes = append(ipCtx.GetExcludedPrefixes(), prefixes...)
+
+	conn, err := next.Server(ctx).Request(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = eps.validateConnection(conn); err != nil {
+		logger.Errorf("ExcludedPrefixesService: connection is invalid: %v", err)
+		_, _ = next.Server(ctx).Close(ctx, conn)
```

Comment:
Do we need to log error here?
```suggestion
		_, err = next.Server(ctx).Close(ctx, conn)
```

Reply:
ok, added